fwutil/lib.py

```diff
@@ -687,8 +791,126 @@ def update_firmware(self, chassis_name, module_name, component_name):
             log_helper.log_fw_update_end(component_path, firmware_path, False, e)
             raise
 
+    def update_au_status_file(self, au_info_data, filename=FW_AU_STATUS_FILE_PATH):
+        with open(filename, 'w') as f:
+            json.dump(au_info_data, f)
+
+    def read_au_status_file_if_exists(self, filename=FW_AU_STATUS_FILE_PATH):
+        data = None
+        if os.path.exists(filename):
+            with open(filename) as au_status_file:
+                data = json.load(au_status_file)
+        return data
+
+    def set_firmware_auto_update_status(self, component_path, boot, rt_code):
+        data = self.read_au_status_file_if_exists(FW_AU_STATUS_FILE_PATH)
+        if data is None:
+            data = {}
+        if boot not in data:
+            data[boot] = []
+
+        au_status = data[boot]
+
+        comp_au_status = {}
+        if rt_code < -1:
+            status = False
+        else:
+            status = True
+
+        if rt_code == 0:
+            info = "reserved"
+        elif rt_code == 1:
+            info = "installed"
+        elif rt_code == 2:
+            info = "updated"
+        elif rt_code == 3:
+            info = "scheduled"
+        elif rt_code == -1:
+            info = "err_boot_type"
+        elif rt_code == -2:
+            info = "err_image"
+        elif rt_code == -3:
+            info = "err_others"
+        else:
+            info = "err_unknown"
+
+        comp_au_status['comp'] = component_path
+        comp_au_status['status'] = status
+        comp_au_status['info'] = info
```

Comment:
Please add version details.

Reply:
@ArunSaravananBalachandran I fixed only the parsing part based on the last HLD update. But the last HLD update is a little gap between the requirement for reboot_handler. I will set up a meeting sometime tomorrow to revise the HLD again.